File: JavaScript/RestaurantAPP.js
//Variables necessaries

const NOTE5 = 5;
const NOTE10 = 10;
const NOTE20 = 20;
const NOTE50 = 50;
const NOTE100 = 100;
const NOTE200 = 200;
const NOTE500 = 500;

var totalcost;

var menu = new Array;
var price = new Array;

// varroduir plats i preus dins d'arrays
var platePrices = new Map;

platePrices.set("Braves", 4.50);
platePrices.set("Xipirons", 6.00);
platePrices.set("Pop", 9.50);
platePrices.set("Xistorra", 5.40);
platePrices.set("Musclos", 8.00);

var k = 0;
platePrices.forEach((prices, plate) => {
    menu[k] = plate;
    price[k] = prices;
    k++;
});

// Mostrar arrays per pantalla

for (let i = 0; i < price.length; i++) {
    document.writeln(menu[i] + " - " + price[i] + " €", "</br>");
}

// Sol·licitud de comanda
var order = new Map;

var keep = 1;
while (keep != 0) {
    plate = prompt("Què voleu menjar?")

    if (order.get(plate) == null) {
        order.set(plate, 1);
    } else {
        order.set(plate, order.get(plate) + 1);
    }

    do {

        keep = prompt("Voleu demanar un altre plat? (1:Si / 0:No)")

        if (keep != 0 && keep != 1) {
            alert("Heu d'introduir 0 o 1!");
        }
    } while (keep != 0 && keep != 1);

}

// Llistat comanda i preu final
totalcost = 0;

document.writeln("La teva comanda és:","</br>")

order.forEach((numeroPlatos, plateOrder) => {
    for (let i = 0; i < numeroPlatos; i++) {							//per si demanen 2 vegades un plat


        if (platePrices.has(plateOrder)) {
            totalcost = totalcost + platePrices.get(plateOrder);

            document.writeln(plateOrder + " - " + platePrices.get(plateOrder) + " €", "</br>");
        } else {
            document.writeln("El producte " + plateOrder + " no existeix!", "</br>");
        }
    }
});

document.writeln("Preu total " + totalcost + " €", "</br>");

document.writeln("Gràcies per la vostra comanda!", "</br>");

File: src/RestaurantAPP.java
import java.util.*;

public class RestaurantAPP {

	public static void main(String[] args) {

		// Variables necessaries
		final int NOTE5 = 5;
		final int NOTE10 = 10;
		final int NOTE20 = 20;
		final int NOTE50 = 50;
		final int NOTE100 = 100;
		final int NOTE200 = 200;
		final int NOTE500 = 500;

		double totalcost;

		String[] menu = new String[5];
		double[] price = new double[5];

		// Introduir plats i preus dins d'arrays
		HashMap<String, Double> platePrices = new HashMap<String, Double>();

		platePrices.put("Braves", 4.50);
		platePrices.put("Xipirons", 6.00);
		platePrices.put("Pop", 9.50);
		platePrices.put("Xistorra", 5.40);
		platePrices.put("Musclos", 8.00);

		int k = 0;
		for (String plate : platePrices.keySet()) {
			menu[k] = plate;
			price[k] = platePrices.get(plate);
			k++;
		}

		// Mostrar arrays per pantalla
		for (int i = 0; i < price.length; i++) {
			System.out.println(menu[i] + " - " + price[i] + " €");
		}

		// Sol·licitud de comanda
		HashMap<String, Integer> order = new HashMap<String, Integer>();

		int keep = 1;
		while (keep != 0) {
			System.out.println("Què voleu menjar?");

			Scanner sc = new Scanner(System.in);
			String plate = sc.next();
			
			if (order.get(plate) == null) {
				order.put(plate, 1);
			} else {
				order.put(plate, order.get(plate) + 1);
			}

			do {
				try {
					System.out.println("Voleu demanar un altre plat? (1:Si / 0:No)");
					
					Scanner sc1 = new Scanner(System.in);
					keep = sc1.nextInt();
				
				} catch (InputMismatchException e) {
					System.out.println("Heu d'introduir 0 o 1!");
					keep = 3;
				}
			} while (keep != 0 && keep != 1);

		}

		// Llistat comanda i preu final
		totalcost = 0;

		for (String plateOrder : order.keySet()) {
			for (int i = 0; i < order.get(plateOrder); i++) {							//per si demanen 2 vegades un plat
				try {
					totalcost = totalcost + platePrices.get(plateOrder);

					System.out.println(plateOrder + " - " + platePrices.get(plateOrder));
				} catch (Exception e) {
					System.out.println("El producte " + plateOrder + " no existeix!");
				}
			}

		}

		System.out.println("Preu total " + totalcost);

		System.out.println("Gràcies per la vostra comanda!");
	}

}
